### PR TITLE
adds a monster I saw in a dream to portal storms

### PR DESCRIPTION
**Pull request by Daved27hundred**
#### Summary
Content "New portal storm monster based off a dream I had"

#### Purpose of change

Like a month ago I had a dream that I was playing CDDA and I got attacked by something that looked like a flat shape that flew above my character like a stingray. I thought of making an actual monster based on it. Took a couple creative liberties with it but I think it fits what I saw in my dream really well.

#### Describe the solution

I interpreted the weird thing I saw into the "Flatlander," A being from a universe with only two spatial dimensions. Standing too close to the flatlander will cause it to momentarily snap your character into two dimensions, stunning them and inducing nausea.

~~I'm going to be honest - I've no clue if I did the monster's spell correctly, which is why I made this a draft PR. I'll try and fix it tomorrow as it's getting late.~~ Guardian fixed the spell for me, big thanks to him.

#### Describe alternatives you've considered

I am open to any changes or suggestions.

#### Additional context

It didn't have a name in my dream so the name, description and the attack were thought up after. I don't recall it ever attacking me in the dream because I stayed away from the entire section of a town it was in.

### CODE REVIEW

**Comment by Zireael07** [comment]
offtop: Flatlander woman

Also: I see I'm not the only one who dreams of playing CDDA (though the monsters I fought were more applicable to Magiclysm instead being standard fantasy fare)

File: data/json/monster_special_attacks/spells.json [hunk 1]
```diff
@@ -975,6 +975,17 @@
...
+    "type": "SPELL",
+    "id": "flatlander_attack",
+    "name": { "str": "flatlander attack" },
+    "description": "dazes the player and makes them nauseous",
+    "valid_targets": [ "hostile" ],
+    "effect": "attack",
+    "effect_str": "stunned",
+    "max_range": 4,
+    "extra_effects": [ { "id": "nausea" } ]
+  },
```
Comment (GuardianDll) [inline data/json/monster_special_attacks/spells.json:988]:
```suggestion
  {
    "type": "SPELL",
    "id": "flatlander_attack",
    "name": { "str": "flatlander attack" },
    "description": "Dazes the player",
    "valid_targets": [ "hostile" ],
    "effect": "attack",
    "effect_str": "stunned",
    "max_range": 4,
    "shape": "blast",
    "flags": [ "SILENT", "NO_EXPLOSION_SFX" ],
    "extra_effects": [ { "id": "flatlander_attack_nausea", "hit_self": true } ]
  },
  {
    "type": "SPELL",
    "id": "flatlander_attack_nausea",
    "name": { "str": "flatlander attack nausea" },
    "description": "Makes enemy nauseous",
    "valid_targets": [ "hostile" ],
    "effect": "attack",
    "shape": "blast",
    "flags": [ "SILENT", "NO_EXPLOSION_SFX" ],
    "effect_str": "nausea"
  }
```

Comment (Daved27hundred) [reply data/json/monster_special_attacks/spells.json:988]: thx guardian

**Comment by bombasticSlacks** [comment]
Needs to be added to EOC_PORTAL_EFFECTS_PASSIVE for it to spawn at all.

**Pull request merged**